### DISCUSSION TITLE
Error in the WebXR implementation of the documentation

### DISCUSSION THREAD

**Discussion by hd-esteban** [google/model-viewer#2320] [Q&A]
the first example of the below page shows the prompt hand until the 3d model has been displayed not during the process of recognizing the space; it seems to be happening because the event changes from not-presenting to session-started until the 3d model has loaded and displayed. So, would be great to have an intermediate event called initializing-session to allow us to distinguish both events properly

https://modelviewer.dev/examples/augmentedreality/

**Comment by elalish** [comment]
That doesn't seem to be the behavior I'm seeing; can you fill out the issue template and include a screen recording so I can be sure I know what you mean?

**Comment by hd-esteban** [comment]
I don't have a way to record a video at this moment, but let me explain what is happening on my end,

- The first time I'm opening the example, and tapping the "View in your space" button.
- Chrome requests some permissions to proceed with webxr.
- Then the camera is enabled, and the hand with the circle/infinite animation is not displayed.
- when I move the device and the 3d model is displayed.
- and until this moment the animation starts in an infinite loop.

I think the animation should be displayed before the model is displayed and removed when the product is loaded.

Animation displayed even if the product has been rendered
![image](https://user-images.githubusercontent.com/1881501/115940415-1e304a00-a467-11eb-9bbf-9b549ccd61fd.png)

Animation not displayed while loading
![image](https://user-images.githubusercontent.com/1881501/115940425-27211b80-a467-11eb-8d13-6f407a718975.png)

**Reply by elalish** [reply]
I believe you've misunderstood the AR placement process: the model is first shown before it has been placed (you must move your phone around and be in a reasonable environment for ARCore to find your floor) and during this time the animation goes. It stops when the floor is found and the object is placed upon it. I'm also in the process of improving this placement flow in #2279.

**Reply by hd-esteban** [reply]
Thank you for your quick support, now it makes sense; the first time the model appears but it is floating in the air, maybe my device is not finding a proper area due to the glass of the table.
